File: daily_challenge_leetcode/2197.py
from math import gcd

class Solution:
    def replaceNonCoprimes(self, nums):
        result = []
        for num in nums:
            while result:
                prev = result[-1]
                curr = num
                GCD = gcd(prev, curr)
                if GCD == 1:
                    break
                result.pop()
                LCM = prev // GCD * curr
                num = LCM
            result.append(num)
        return num

# from math import gcd
# class Solution:
#     def replaceNonCoprimes(self, nums):
#         result = []
#         for num in nums:
#             while result:
#                 prev = result[-1]
#                 curr = num
#                 GCD = gcd(prev, num)
#                 if GCD == 1:
#                     break
#                 result.pop()
#                 LCM = prev // GCD * curr
#                 num = LCM
#             result.append(num)
#         return num

File: daily_challenge_leetcode/2197.js
function gcd(a, b) {
    while (b !== 0) {
        let temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}
var replaceNonCoprimes = function(nums) {
    const result = [];
    for (let num of nums) {
        while (result.length > 0) {
            let prev = result[result.length - 1];
            let curr = num;
            let GCD = gcd(prev, curr);
            if (GCD === 1) {
                break;
            }
            let LCM = (prev / GCD) * curr;
            num = LCM;
            result.pop();
        }
        result.push(num);
    }
    return result;
};